src/pathfora.js

```diff
@@ -2261,97 +2279,226 @@
-          var matched = false,
-              defaultEl = {};
+      attrs.forEach(function (attr) {
+        var elements = inline.prepElements(attr);
 
-          for (var i = 0; i < elementObj[key].length; i++) {
-            var singleElementObj = elementObj[key][i];
+        for (var key in elements) {
+          if (elements.hasOwnProperty(key)) {
 
-            // if we find a match show that and prevent others from showing in same group
-            if (inSegment(singleElementObj.trigger) && !matched) {
-              singleElementObj.elem.removeAttribute('data-pftrigger');
-              singleElementObj.elem.setAttribute('data-pfmodified', 'true');
+            switch (attr) {
+            // CASE: Segment triggered elements
+            case 'data-pftrigger':
+              inline.procTriggerElements(elements[key], key);
+              break;
 
-              if (key !== 'default') {
-                matched = true;
-                continue;
-              }
-            }
-
-            // if this is the default save it
-            if (singleElementObj.trigger === 'default') {
-              defaultEl = singleElementObj;
+            // CASE: Content recommendation elements
+            case 'data-pfrecommend':
+              inline.procRecommendElements(elements[key], key, function () {
+                cb(elements);
+              });
+              break;
             }
           }
+        }
+      });
+    };
+
+    this.procTriggerElements = function (elems, group) {
+      var matched = false,
+          defaultEl = {};
+
+      for (var i = 0; i < elems.length; i++) {
+        var elem = elems[i];
 
-          // if nothing matched show the default
-          if (!matched && key !== 'default' && defaultEl.elem) {
-            defaultEl.elem.removeAttribute('data-pftrigger');
-            defaultEl.elem.setAttribute('data-pfmodified', 'true');
+        // if we find a match show that and prevent others from showing in same group
+        if (api.inSegment(elem.trigger) && !matched) {
+          elem.elem.removeAttribute('data-pftrigger');
+          elem.elem.setAttribute('data-pfmodified', 'true');
+          this.preppedElements[group] = elem;
+
+          if (group !== 'default') {
+            matched = true;
+            continue;
           }
         }
+
+        // if this is the default save it
+        if (elem.trigger === 'default') {
+          defaultEl = elem;
+        }
       }
 
-      this.preppedElements = elementObj;
+      // if nothing matched show default
+      if (!matched && group !== 'default' && defaultEl.elem) {
+        defaultEl.elem.removeAttribute('data-pftrigger');
+        defaultEl.elem.setAttribute('data-pfmodified', 'true');
+        this.preppedElements[group] = defaultEl;
+      }
     };
 
+    this.procRecommendElements = function (blocks, rec, cb) {
+      var inline = this;
+
+      if (rec !== 'default') {
+        // TODO: NEED TO MAKE SURE FILTER IS WORKING, SEEMS BROKEN ATM
```

Comment:
Whoops yeah, this actually isn't even broken anymore. 😁 So I'm just gonna remove this comment.